python/mxnet/gluon/block.py

```diff
@@ -932,6 +932,9 @@ def _get_graph(self, *args):
 
     def _build_cache(self, *args):
         data, out = self._get_graph(*args)
+        if self._backend:
+            # To do: pass in all arguments
+            out = out.optimize_for(self._backend, **self.optargs)
```

Comment:
can we add a comment that we're partitioning the graph here?

Reply:
Added.

Reply:
Why not do it inside `self._get_graph`?

Reply:
agreed, lets move it here:
https://github.com/apache/incubator-mxnet/blob/557a29b7024e86979b5927ae81209d8c32d8511d/python/mxnet/gluon/block.py#L927-L928

Reply:
Just as Sam mentioned "Because once self._cached_graph is set the previous code wont be executed in the _get_graph function". It is better to keep this change.